src/MPAGSCipher/RunCaesarCipher.cpp

```diff
@@ -0,0 +1,116 @@
+#include "RunCaesarCipher.hpp"
+
+std::string transform_char(const char in_char);
+
+std::string runCaesarCipher(const std::string& inputText, size_t key, const bool encrypt)
+{
+    // Create the alphabet container and output string
+    const std::string alphabet{"ABCDEFGHIJKLMNOPQRSTUVWXYZ"};
+    std::string outputText{""};
+    std::string transformed_str{""};
+
+    // Transform the input text: convert all letters to capitals and replace numbers with letters
+    for (int i = 0; i < int(inputText.length()); i++)
+    {   
+        char character{inputText[i]};
+        transformed_str += transform_char(character);
+    }
+
+    for (int i = 0; i < int(transformed_str.length()); i++)
+    {   
+        // Loop over the transformed string
+        char character{transformed_str[i]};
+        int index{-1};
+
+        // For each character find the corresponding position in the alphabet
+        index = alphabet.find(character);
+        
+        // For each character find the corresponding position in the alphabet
+        int new_index{-1};
+        
+        // Apply the shift (depending on encrypt/decrypt value)
+        if (encrypt) {
+            new_index = index + key;
+            // Handling wrap-around for encryption
+            if (new_index > int(alphabet.length()-1)) {
+                new_index = new_index % alphabet.length();
+            }
+        }
+        else if (!encrypt) {
+            new_index = index - key;
+            // Handling wrap-around for decryption
+            if (new_index < 0) {
+                new_index = alphabet.length() + new_index;
+            }
```

Comment:
Again can simplify things to remove the `if` and also fix a bug:
This time because you're subtracting the key you need to ensure it's within 0 - 25 to avoid `alphabet.length() + index - key` from potentially becoming negative.  As currently written, the decryption will potentially start to corrupt some characters for key values > 26.

Here's the fix alone:
```suggestion
            new_index = index - (key%alphabet.length());
            // Handling wrap-around for decryption
            if (new_index < 0) {
                new_index = alphabet.length() + new_index;
            }
```

And here's the additional simplification:
```suggestion
            // Handling wrap-around for decryption
            new_index = (alphabet.length() + index - (key%alphabet.length()) ) % alphabet.length();
```

Now, for efficiency reasons and also to make the code more readable you really want to store `alphabet.length()` and `key % alphabet.length()` in const variables before the loop begins.